server/base/src/main/java/org/apache/accumulo/server/util/MasterMetadataUtil.java

```diff
@@ -227,14 +204,6 @@ public static StoredTabletFile updateTabletDataFile(ServerContext context, KeyEx
       tablet.putFile(path, dfv);
       tablet.putTime(time);
       newFile = path.insert();
-
-      TServerInstance self = getTServerInstance(address, zooLock);
-      tablet.putLocation(self, LocationType.LAST);
-
-      // remove the old location
-      if (lastLocation != null && !lastLocation.equals(self)) {
-        tablet.deleteLocation(lastLocation, LocationType.LAST);
-      }
```

Comment:
Why is it the case that these location entries no longer need to be updated here? Where is the replacement code?

Reply:
The replacement code is my changes in #1616. That is what spawned most of these things. Though maybe this is still considered necessary but if lastLocation is properly set through those other changes then it doesn't have to be set here. I believe after #1616 this function wasn't used anymore but it has been awhile, I can go back and confirm that aspect.

Reply:
Okay, that makes sense. I tried to trace the callers to this method, to see if they instead ended up reaching the new code path in #1616, but got tired last night. I'll look at it again today. I just want to make sure that if we're not updating it here, that it's because we definitely don't need to.